client/clientservice/src/request_service.cpp

```diff
@@ -120,7 +125,13 @@ void RequestServiceCallData::sendToConcordClient() {
     auto reply = std::get<bft::client::Reply>(send_result);
     // We need to copy because there is no implicit conversion between vector<uint8> and std::string
     std::string data(reply.matched_data.begin(), reply.matched_data.end());
-    this->response_.set_response(std::move(data));
+
+    // Check if the application response is of Any Type then set it to Any response.
+    google::protobuf::Any* anyResponse = this->response_.mutable_typed_response();
+    if (anyResponse->ParseFromString(data) == false) {
```

Comment:
This is not a reliable way to determine a type of a message. We either need to switch to serializing a container message (such as `google.protobuf.Any`) or assume for now that the response is always going to be of a certain type (such as `ConcordResponse` or `DamlResponse`).
I'd strongly suggest doing the latter first, and then changing the protocol between the BFT client and the replicas to contain the actual type of the messages traveling.